src/util/StreamLink.cpp

```diff
@@ -165,15 +179,31 @@ void getStreamQualities(const QString &channelURL,
             }
         });
 
-    p->setArguments({channelURL, "--default-stream=KKona"});
+    p->setArguments(p->arguments() +
+                    QStringList({channelURL, "--default-stream=KKona"}));
 
     p->start();
 }
 
 void openStreamlink(const QString &channelURL, const QString &quality,
                     QStringList extraArguments)
 {
-    QStringList arguments = extraArguments << channelURL << quality;
+    QStringList arguments;
+
+    QString program;
+
+    if (Version::instance().isFlatpak())
+    {
+        program = "flatpak-spawn";
+
+        arguments << "--host" << getStreamlinkProgram();
+    }
+    else
+    {
+        program = getStreamlinkProgram();
+    }
```

Comment:
Maybe makes the code a little less readable, but I feel like we can get rid of this else at the end.
```suggestion
    QString program = getStreamlinkProgram();

    if (Version::instance().isFlatpak())
    {
        program = "flatpak-spawn";

        arguments << "--host" << getStreamlinkProgram();
    }
```